corehq/apps/userreports/reports/view.py

```diff
@@ -668,18 +664,27 @@ def section_url(self):
         return "#"
 
     def get(self, request, *args, **kwargs):
-        context = super(DownloadUCRStatusView, self).main_context
-        context.update({
-            'domain': self.domain,
-            'download_id': kwargs['download_id'],
-            'poll_url': reverse('ucr_download_job_poll', args=[self.domain, kwargs['download_id']]),
-            'title': _("Download Report Status"),
-            'progress_text': _("Preparing report download."),
-            'error_text': _("There was an unexpected error! Please try again or report an issue."),
-            'next_url': reverse(ConfigurableReport.slug, args=[self.domain, self.report_config_id]),
-            'next_url_text': _("Go back to report"),
-        })
-        return render(request, 'hqwebapp/soil_status_full.html', context)
+        if _has_permission(self.domain, request.couch_user, self.report_config_id):
+            context = super(DownloadUCRStatusView, self).main_context
+            context.update({
+                'domain': self.domain,
+                'download_id': kwargs['download_id'],
+                'poll_url': reverse('ucr_download_job_poll',
+                                    args=[self.domain, kwargs['download_id']],
+                                    params={'config_id': self.report_config_id}),
+                'title': _("Download Report Status"),
+                'progress_text': _("Preparing report download."),
+                'error_text': _("There was an unexpected error! Please try again or report an issue."),
+                'next_url': reverse(ConfigurableReport.slug, args=[self.domain, self.report_config_id]),
+                'next_url_text': _("Go back to report"),
+            })
+            return render(request, 'hqwebapp/soil_status_full.html', context)
+        else:
+            raise Http403()
+
+    @conditionally_location_safe(has_location_filter)
+    def dispatch(self, *args, **kwargs):
+        return super(DownloadUCRStatusView, self).dispatch(*args, **kwargs)
```

Comment:
I think you should be able to decorate the class itself, yeah?

Reply:
Hmm, actually it looks like `@conditionally_location_safe` doesn't do that - it would be pretty simple to add (see how `@location_safe` does it), but I'll leave it up to you.